Source/src/components/ComponentAgent.cpp

```diff
@@ -12,6 +12,10 @@
 
 Hachiko::ComponentAgent::ComponentAgent(GameObject* container) : Component(Type::AGENT, container)
 {
+    if (game_object->name == "PlayerN")
+    {
+        player = true;
+    }
```

Comment:
We shouldn't have this hard-coded by the name of the game object, engine should be as much as abstracted from the game it runs. Let's make a method of component agent that is: ComponentAgent::SetAsPlayer(bool is_player),  and we set the flag accordingly. If anything needs to be rebuilt based on is_player flag, we can set the appropriate variables based on the flag passed to SetAsPlayer.